core/node/proof_data_handler/src/tests.rs

```diff
@@ -0,0 +1,247 @@
+// 1. Mocking an object store with a single batch blob containing TEE verifier input
+// 2. Populating the SQL db with relevant information about the status of the TEE verifier input and
+//    TEE proof generation
+// 3. Sending a request to the /tee_proof_inputs endpoint and asserting that the response
+//    matches the file from the object store
+#[tokio::test]
+async fn request_tee_proof_inputs() {
+    // prepare a sample mocked TEE verifier input
+
+    let batch_number = L1BatchNumber::from(1);
+    let tvi = TeeVerifierInput::new(
+        PrepareBasicCircuitsJob::new(0),
+        vec![],
+        L1BatchEnv {
+            previous_batch_hash: Some(H256([1; 32])),
+            number: batch_number,
+            timestamp: 0,
+            fee_input: Default::default(),
+            fee_account: Default::default(),
+            enforced_base_fee: None,
+            first_l2_block: L2BlockEnv {
+                number: 0,
+                timestamp: 0,
+                prev_block_hash: H256([1; 32]),
+                max_virtual_blocks_to_create: 0,
+            },
+        },
+        SystemEnv {
+            zk_porter_available: false,
+            version: Default::default(),
+            base_system_smart_contracts: BaseSystemContracts {
+                bootloader: SystemContractCode {
+                    code: vec![U256([1; 4])],
+                    hash: H256([1; 32]),
+                },
+                default_aa: SystemContractCode {
+                    code: vec![U256([1; 4])],
+                    hash: H256([1; 32]),
+                },
+            },
+            bootloader_gas_limit: 0,
+            execution_mode: TxExecutionMode::VerifyExecute,
+            default_validation_computational_gas_limit: 0,
+            chain_id: Default::default(),
+        },
+        vec![(H256([1; 32]), vec![0, 1, 2, 3, 4])],
+    );
+
+    // populate mocked object store with a single batch blob
+
+    let blob_store = ObjectStoreFactory::mock().create_store().await;
+    let object_path = blob_store.put(batch_number, &tvi).await.unwrap();
+
+    // get connection to the SQL db and mock the status of the TEE proof generation
+
+    let db_conn_pool = ConnectionPool::test_pool().await;
+    mock_tee_batch_status(db_conn_pool.clone(), batch_number, &object_path).await;
+
+    // test the /tee_proof_inputs endpoint; it should return the batch from the object store
+
+    let app = create_proof_processing_router(
+        blob_store,
+        db_conn_pool,
+        ProofDataHandlerConfig {
+            http_port: 1337,
+            proof_generation_timeout_in_secs: 10,
+        },
+        L1BatchCommitmentMode::Rollup,
+    );
+    let req_body = Body::from(serde_json::to_vec(&json!({})).unwrap());
+    let response = app
+        .oneshot(
+            Request::builder()
+                .method(Method::POST)
+                .uri("/tee_proof_inputs")
+                .header(http::header::CONTENT_TYPE, mime::APPLICATION_JSON.as_ref())
```

Comment:
I think a string `"application/json"` would be as declarative, and would let us avoid a direct dependency on `mime`.

Reply:
[Fixed](https://github.com/matter-labs/zksync-era/pull/1993/commits/79611757437dfa02095ff7fede8251bd6efa6572) ✅